src/audio/copier/copier_dai.c

```diff
@@ -265,7 +265,7 @@ int copier_dai_create(struct comp_dev *dev, struct copier_data *cd,
 
 	/* create multi_endpoint_buffer for ALH multi-gateway case */
 	if (dai_count > 1) {
-		ret = create_endpoint_buffer(dev, cd, config, copier, type, true, 0);
```

Comment:
@btian it will be better if you split this PR into 2 , one with the copier clean up and the other for using the update_buffer_format function for the other modules

Reply:
outdated, all changes should be done now.